src/test/java/net/bhardy/bizzo/billing/PolicyBuilderTest.java

```diff
@@ -33,233 +33,197 @@
                 .build();
 
-        LocalDate oneSaturday = LocalDate.of(2017, 8, 5);
-
-        List<LocalDate> nextBills = policy.upcomingDueDates(oneSaturday, 2);
-
-        LocalDate expected1 = LocalDate.of(2017, 8, 7);
-        LocalDate actual1 = nextBills.get(0);
-        assertEquals(expected1, actual1);
-        assertEquals(MONDAY, actual1.getDayOfWeek());
+        List<LocalDate> nextBills = policy.upcomingDueDates(saturday5th, 2);
 
-        LocalDate expected2 = LocalDate.of(2017, 8, 8);
-        LocalDate actual2 = nextBills.get(1);
-        assertEquals(expected2, actual2);
-        assertEquals(TUESDAY, actual2.getDayOfWeek());
+        assertEquals(monday7th, nextBills.get(0));
+        assertEquals(tuesday8th, nextBills.get(1));
     }
 
     @Test
     public void upcomingDueDates_weeklySimple() {
+        LocalDate friday11th = friday4th.plusWeeks(1);
+
         BillingPolicy policy = BillingPolicy.builder()
-                .sameDayEveryWeek(FRIDAY)
+                .weeklyOnDay(FRIDAY)
                 .build();
+        List<LocalDate> nextBills = policy.upcomingDueDates(friday4th, 2);
 
-        LocalDate itsFriday = LocalDate.of(2017, 8, 4);
-        List<LocalDate> nextBills = policy.upcomingDueDates(itsFriday, 2);
-
-        LocalDate expected1 = LocalDate.of(2017, 8, 4);
-        LocalDate actual1 = nextBills.get(0);
-        assertEquals(expected1, actual1);
-        assertEquals(FRIDAY, actual1.getDayOfWeek());
-
-        LocalDate expected2 = LocalDate.of(2017, 8, 11);
-        LocalDate actual2 = nextBills.get(1);
-        assertEquals(expected2, actual2);
-        assertEquals(FRIDAY, actual2.getDayOfWeek());
+        assertEquals(friday4th, nextBills.get(0));
+        assertEquals(friday11th, nextBills.get(1));
     }
 
     @Test
     public void upcomingDueDates_weeklyFiltered() {
         BillingPolicy policy = BillingPolicy.builder()
-                .sameDayEveryWeek(SATURDAY)
+                .weeklyOnDay(SATURDAY)
                 .filter(not(daysOfWeek(SUNDAY)))
                 .action(NEXT_DAY)
                 .build();
+        List<LocalDate> nextBills = policy.upcomingDueDates(saturday5th, 2);
 
-        LocalDate oneSaturday = LocalDate.of(2017, 8, 5);
-
-        List<LocalDate> nextBills = policy.upcomingDueDates(oneSaturday, 2);
-
-        LocalDate expected1 = LocalDate.of(2017, 8, 5);
-        LocalDate actual1 = nextBills.get(0);
-        assertEquals(expected1, actual1);
-        assertEquals(SATURDAY, actual1.getDayOfWeek()); // just to be obvious
+        assertEquals(saturday5th, nextBills.get(0));
 
-        LocalDate expected2 = LocalDate.of(2017, 8, 12);
-        LocalDate actual2 = nextBills.get(1);
-        assertEquals(expected2, actual2);
-        assertEquals(SATURDAY, actual2.getDayOfWeek());
+        LocalDate saturday12th = LocalDate.of(2017, 8, 12);
+        assertEquals(saturday12th, nextBills.get(1));
     }
 
     @Test
     public void upcomingDueDates_monthlySimple() {
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Rename this method name to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS00100)